### ISSUE TITLE
When playback ends, panic!

### ISSUE THREAD

**Issue by myyrakle** [RustAudio/rodio#486]
Version 0.17.1
  
In the following form, I wrote a program that stops the currently playing sound and runs the next sound.
```
sink.stop();
while !sink.empty() {}
sink.append(new_source);
sink.play();
...
// music quit
```

However, when the replaced sound is completed, the following error occurs.
```
thread 'cpal_wasapi_out' panicked at 'index out of bounds: the len is 0 but the index is 0', C:\Users\sssan\Code\Rust\rodio\src\decoder\symphonia.rs:175:22
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/658fad6c5506f41c35b64fb1a22ceb0992697ff3/library\std\src\panicking.rs:575
   1: core::panicking::panic_fmt
             at /rustc/658fad6c5506f41c35b64fb1a22ceb0992697ff3/library\core\src\panicking.rs:64
   2: core::panicking::panic_bounds_check
             at /rustc/658fad6c5506f41c35b64fb1a22ceb0992697ff3/library\core\src\panicking.rs:159
   3: rodio::decoder::symphonia::impl$2::next
             at C:\Users\sssan\Code\Rust\rodio\src\decoder\symphonia.rs:175
   4: rodio::decoder::impl$4::next<std::io::buffered::bufreader::BufReader<std::fs::File> >
             at C:\Users\sssan\Code\Rust\rodio\src\decoder\mod.rs:274
   5: rodio::source::speed::impl$1::next<rodio::decoder::Decoder<std::io::buffered::bufreader::BufReader<std::fs::File> > >
             at C:\Users\sssan\Code\Rust\rodio\src\source\speed.rs:56
   6: rodio::source::pausable::impl$1::next<rodio::source::speed::Speed<rodio::decoder::Decoder<std::io::buffered::bufreader::BufReader<std::fs::File> > > >
             at C:\Users\sssan\Code\Rust\rodio\src\source\pausable.rs:85
   7: rodio::source::amplify::impl$1::next<rodio::source::pausable::Pausable<rodio::source::speed::Speed<rodio::decoder::Decoder<std::io::buffered::bufreader::BufReader<std::fs::File> > > > >
             at C:\Users\sssan\Code\Rust\rodio\src\source\amplify.rs:56
   8: rodio::source::skippable::impl$1::next<rodio::source::amplify::Amplify<rodio::source::pausable::Pausable<rodio::source::speed::Speed<rodio::decoder::Decoder<std::io::buffered::bufreader::BufReader<std::fs::File> > > > > >
             at C:\Users\sssan\Code\Rust\rodio\src\source\skippable.rs:58
   9: rodio::source::stoppable::impl$1::next<rodio::source::skippable::Skippable<rodio::source::amplify::Amplify<rodio::source::pausable::Pausable<rodio::source::speed::Speed<rodio::decoder::Decoder<std::io::buffered::bufreader::BufReader<std::fs::File> > > > >
             at C:\Users\sssan\Code\Rust\rodio\src\source\stoppable.rs:57
  10: rodio::source::periodic::impl$1::next<rodio::source::stoppable::Stoppable<rodio::source::skippable::Skippable<rodio::source::amplify::Amplify<rodio::source::pausable::Pausable<rodio::source::speed::Speed<rodio::decoder::Decoder<std::io::buffered::bufreade
             at C:\Users\sssan\Code\Rust\rodio\src\source\periodic.rs:86
  11: rodio::source::samples_converter::impl$1::next<rodio::source::periodic::PeriodicAccess<rodio::source::stoppable::Stoppable<rodio::source::skippable::Skippable<rodio::source::amplify::Amplify<rodio::source::pausable::Pausable<rodio::source::speed::Speed<ro
             at C:\Users\sssan\Code\Rust\rodio\src\source\samples_converter.rs:56
  12: rodio::source::done::impl$1::next<rodio::source::samples_converter::SamplesConverter<rodio::source::periodic::PeriodicAccess<rodio::source::stoppable::Stoppable<rodio::source::skippable::Skippable<rodio::source::amplify::Amplify<rodio::source::pausable::P
             at C:\Users\sssan\Code\Rust\rodio\src\source\done.rs:53
  13: alloc::boxed::impl$39::next<dyn$<rodio::source::Source<assoc$<Item,f32> >,core::marker::Send>,alloc::alloc::Global>
             at /rustc/658fad6c5506f41c35b64fb1a22ceb0992697ff3\library\alloc\src\boxed.rs:1922
  14: rodio::queue::impl$2::next<f32>
             at C:\Users\sssan\Code\Rust\rodio\src\queue.rs:178
  15: rodio::source::uniform::impl$3::next<rodio::queue::SourcesQueueOutput<f32> >
             at C:\Users\sssan\Code\Rust\rodio\src\source\uniform.rs:159
  16: rodio::conversions::sample_rate::SampleRateConverter<rodio::source::uniform::Take<rodio::queue::SourcesQueueOutput<f32> > >::next_input_frame<rodio::source::uniform::Take<rodio::queue::SourcesQueueOutput<f32> > >
             at C:\Users\sssan\Code\Rust\rodio\src\conversions\sample_rate.rs:111
  17: rodio::conversions::sample_rate::impl$1::next<rodio::source::uniform::Take<rodio::queue::SourcesQueueOutput<f32> > >
             at C:\Users\sssan\Code\Rust\rodio\src\conversions\sample_rate.rs:160
  18: rodio::conversions::channels::impl$1::next<rodio::conversions::sample_rate::SampleRateConverter<rodio::source::uniform::Take<rodio::queue::SourcesQueueOutput<f32> > > >
             at C:\Users\sssan\Code\Rust\rodio\src\conversions\channels.rs:62
  19: rodio::conversions::sample::impl$1::next<rodio::conversions::channels::ChannelCountConverter<rodio::conversions::sample_rate::SampleRateConverter<rodio::source::uniform::Take<rodio::queue::SourcesQueueOutput<f32> > > >,f32>
             at C:\Users\sssan\Code\Rust\rodio\src\conversions\sample.rs:38
  20: rodio::source::uniform::impl$1::next<rodio::queue::SourcesQueueOutput<f32>,f32>
             at C:\Users\sssan\Code\Rust\rodio\src\source\uniform.rs:88
  21: alloc::boxed::impl$39::next<dyn$<rodio::source::Source<assoc$<Item,f32> >,core::marker::Send>,alloc::alloc::Global>
             at /rustc/658fad6c5506f41c35b64fb1a22ceb0992697ff3\library\alloc\src\boxed.rs:1922
  22: rodio::dynamic_mixer::DynamicMixer<f32>::sum_current_sources<f32>
             at C:\Users\sssan\Code\Rust\rodio\src\dynamic_mixer.rs:170
  23: rodio::dynamic_mixer::impl$2::next<f32>
             at C:\Users\sssan\Code\Rust\rodio\src\dynamic_mixer.rs:125
  24: rodio::stream::impl$11::new_output_stream_with_format::closure$1::closure$0
             at C:\Users\sssan\Code\Rust\rodio\src\stream.rs:219
  25: core::slice::iter::impl$187::for_each<f32,rodio::stream::impl$11::new_output_stream_with_format::closure$1::closure_env$0>
             at /rustc/658fad6c5506f41c35b64fb1a22ceb0992697ff3\library\core\src\slice\iter\macros.rs:201
  26: rodio::stream::impl$11::new_output_stream_with_format::closure$1
             at C:\Users\sssan\Code\Rust\rodio\src\stream.rs:218
  27: cpal::traits::DeviceTrait::build_output_stream::closure$0<cpal::platform::platform_impl::Device,f32,rodio::stream::impl$11::new_output_stream_with_format::closure_env$1,rodio::stream::impl$11::new_output_stream_with_format::closure_env$0>
             at C:\Users\sssan\.cargo\registry\src\github.com-1ecc6299db9ec823\cpal-0.15.0\src\traits.rs:169
  28: cpal::host::wasapi::stream::process_output
             at C:\Users\sssan\.cargo\registry\src\github.com-1ecc6299db9ec823\cpal-0.15.0\src\host\wasapi\stream.rs:466
  29: cpal::host::wasapi::stream::run_output
             at C:\Users\sssan\.cargo\registry\src\github.com-1ecc6299db9ec823\cpal-0.15.0\src\host\wasapi\stream.rs:307
  30: cpal::host::wasapi::stream::impl$1::new_output::closure$0<cpal::traits::DeviceTrait::build_output_stream::closure_env$0<cpal::platform::platform_impl::Device,f32,rodio::stream::impl$11::new_output_stream_with_format::closure_env$1,rodio::stream::impl$11::
             at C:\Users\sssan\.cargo\registry\src\github.com-1ecc6299db9ec823\cpal-0.15.0\src\host\wasapi\stream.rs:139
```
It doesn't just happen occasionally, it happens over and over again.

**Comment by est31** [comment]
Fixed in #487

**Comment by Mcfloy** [comment]
I've tested the fix with the git option (the 0.17.1 does not contain this fix) and although there's no more panic, the sounds are not being played.

**Comment by Mcfloy** [comment]
I tested with the latest version `0.17.3` (cpal version: `0.15.2`) and it works (for anyone who would have know if it was fixed or not)
